ui/components/src/Legend/ListLegendItem.tsx

```diff
@@ -17,9 +17,16 @@ import { LegendItem } from '../model';
 import { combineSx } from '../utils';
 import { LegendColorBadge } from './LegendColorBadge';
 
-interface ListLegendItemProps extends ListItemProps<'div'> {
+export interface ListLegendItemProps extends Omit<ListItemProps<'div'>, 'onClick'> {
   item: LegendItem;
 
+  /**
+   * When true, the item is visually highlighted to communicate it is selected.
+   */
+  isHighlighted?: boolean;
```

Comment:
Calling this `isHighlighted` instead of `isSelected` because in the "ALL" case, everything is technically selected but is NOT visually highlighted as such.

Reply:
The only thing that might be confusing is a few people have asked about hovering on legend items, highlighting the corresponding line in the chart.

Once the legend is moved inside LineChart.tsx, that could be possible (since the legend would have access to the chart ref) by dispatching a [highlight](https://echarts.apache.org/en/api.html#action.highlight) action similar to the tooltip.

In that case highlighting === emphasized, so all the terms get a little confusing (there is also "focused" in the tooltip for the series that are nearby the cursor 😂 ) .

Reply:
That's a good callout. I could rename this to `isVisuallySelected` which is a little more verbose, but more explicit about what it is and reduces overlap with the terminology used in echarts.

Reply:
done!

Reply:
That sounds great!